TP 2 Commit

maximp14 opened the pull request:
PR from branch dev to master

MaxPLP committed:
TP 2 Commit

My first commit

PedroAguiar reviewed (approved):
Muy buen laburo! Te deje un par de comentarios pero overall very good job!  Fijate que cuando usas H2, la palabra "order" es una palabra reservada de H2 asi que cambia "order" por "pedido" en tu nombre de tabla y foreign keys y join column

PedroAguiar commented on arreglo/src/main/java/com/globant/controller/ClientController.java:
```diff
@@ -0,0 +1,45 @@
+    }
+
+    @PutMapping("/client/{id}")
+    public void updateClient(@PathVariable Integer id,@RequestBody Client client){
```
use DTO's

PedroAguiar commented on arreglo/src/main/java/com/globant/controller/ItemController.java:
```diff
@@ -0,0 +1,48 @@
+    }
+
+    @PutMapping("/order/orderId/item/{id}")
+    public void updateItem(@RequestBody Item item,@PathVariable Integer orderId,@PathVariable Integer id){
```
use DTO's

PedroAguiar commented on arreglo/src/main/java/com/globant/controller/OrderController.java:
```diff
@@ -0,0 +1,50 @@
+
+    @PutMapping("/order/{id}")
+    public void updateOrder(@RequestBody Order order,@PathVariable Integer id){
+        orderService.updateOrder(order,id);
```
use DTO's

PedroAguiar commented on arreglo/src/main/java/com/globant/controller/PaymentController.java:
```diff
@@ -0,0 +1,44 @@
+
+    @PutMapping("/payment/{id}")
+    public void updatePayment(@PathVariable Integer id, @RequestBody Payment payment){
+        paymentService.updatePayment(id, payment);
```
Use DTO's

PedroAguiar commented on arreglo/src/main/java/com/globant/model/Item.java:
```diff
@@ -0,0 +1,51 @@
+    private Order order;
+
+
+    public Item(Integer id, String name) {
```
remove all geters and setters use the @Data from lombok and use the @NoArgsConstructor & @AllArgsConstructor aswell

PedroAguiar commented on arreglo/src/main/java/com/globant/model/Payment.java:
```diff
@@ -0,0 +1,37 @@
+    private Integer id;
+    private float amount;
+
+    public Payment() {
```
remove all geters and setters use the @DaTa from lombok and use the @NoArgsConstructor & @AllArgsConstructor aswell

PedroAguiar commented on arreglo/src/main/java/com/globant/model/Order.java:
```diff
@@ -0,0 +1,38 @@
+    private Integer id;
+    private String name;
+
+    public Order(Integer id, String name) {
```
remove all geters and setters use the @DaTa from lombok and use the @NoArgsConstructor & @AllArgsConstructor aswell

PedroAguiar commented on arreglo/src/main/java/com/globant/model/Payment.java:
```diff
@@ -0,0 +1,37 @@
+    @GeneratedValue(strategy = GenerationType.AUTO)
+    private Integer id;
+    private float amount;
+
```
Payment should have a @OneToOne relationship with Order, also declare a @JoinColumn with a foreignKey definition and a referenced column name

PedroAguiar commented on arreglo/src/main/java/com/globant/model/Order.java:
```diff
@@ -0,0 +1,38 @@
+    @GeneratedValue (strategy = GenerationType.AUTO)
+    private Integer id;
+    private String name;
+
```
Order should have a @ManyToOne relationship with Items and also a @JoinColumn with a foreign key defintion and a referenced column name

PedroAguiar commented on arreglo/src/main/java/com/globant/repository/ItemRepository.java:
```diff
@@ -0,0 +1,11 @@
+
+public interface ItemRepository extends CrudRepository<Item,Integer>{
+
+    List<Item> findByOrderId(Integer orderId);
```
this is provided by default when you etend from JpaRepository insted of CrudRepository

PedroAguiar commented on arreglo/src/main/java/com/globant/repository/PaymentRepository.java:
```diff
@@ -0,0 +1,7 @@
+import com.globant.model.Payment;
+import org.springframework.data.repository.CrudRepository;
+
+public interface PaymentRepository extends CrudRepository<Payment,Integer> {
```
extend from JpaRepository not CrudRepository

PedroAguiar commented on arreglo/src/main/java/com/globant/service/ClientServiceImp.java:
```diff
@@ -0,0 +1,48 @@
+    @Override
+    public List<Client> getAllClients() {
+        List<Client> clients = new ArrayList<>();
+        clientRepository.findAll().forEach(clients::add);
```
once you extend from JpaRepository you will be able to do the follwoing: 
````
return clientRepository.findAll();
````

PedroAguiar commented on arreglo/src/main/java/com/globant/service/ItemServiceImp.java:
```diff
@@ -0,0 +1,50 @@
+    @Override
+    public List<Item> getAllItems(Integer orderId) {
+        List<Item> items = new ArrayList<>();
+        itemRepository.findByOrderId(orderId).forEach(items::add);
```
once you extend from JpaRepository you will be able to do the follwoing: 
````
return itemRepository.findAll();
````

PedroAguiar commented on arreglo/src/main/java/com/globant/service/OrderServiceImp.java:
```diff
@@ -0,0 +1,51 @@
+    @Override
+    public List<Order> getAllOrders(){
+        List<Order> orders = new ArrayList<>();
+        orderRepository.findAll().forEach(orders :: add);
```
once you extend from JpaRepository you will be able to do the follwoing: 
````
return orderRepository.findAll();
````

PedroAguiar commented on arreglo/src/main/java/com/globant/service/PaymentServiceImp.java:
```diff
@@ -0,0 +1,47 @@
+    @Override
+    public List<Payment> getAllPayments() {
+        List<Payment> payments = new ArrayList<>();
+        paymentRepository.findAll().forEach(payments::add);
```
once you extend from JpaRepository you will be able to do the follwoing: 
````
return paymentRepository.findAll();
````

PedroAguiar reviewed (changes requested):
Overall good code, start with the testing as soon as posible

MaxPLP committed:
here goes, i didnt much (i corrected the things you wrote in the review) but over the weekend i'll complete some things
